pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -531,6 +531,9 @@ func (l *LibvirtDomainManager) asyncMigrate(vmi *v1.VirtualMachineInstance, opti
 }
 
 func setupMigration(l *LibvirtDomainManager, vmi *v1.VirtualMachineInstance, dom cli.VirDomain, options *cmdclient.MigrationOptions, loopbackAddress string) (*libvirt.DomainMigrateParameters, error) {
+	l.domainModifyLock.Lock()
+	defer l.domainModifyLock.Unlock()
```

Comment:
Why `getDiskTargetsForMigration` is not part of the lock? Seems it reads the domain as well.
Same for `MigrateToURI3`.

Reply:
> Why `getDiskTargetsForMigration` is not part of the lock? Seems it reads the domain as well.

It is included in the lock.

> Same for `MigrateToURI3`.

I will explain in the commit message.
We cannot do it because of the go-routine that monitors the migration progress, it uses the same lock.

> Why not looking before `LookupDomainByName` and releasing the lock on the `defer` of the go routine?

It is unsafe to lock in one place and release in another (a different abstraction layer, different go-routine) and I think it would not have worked in this specific case (see the answer before about the monitor go-routine).